src/main/java/in/org/projecteka/hiu/consent/ConsentService.java

```diff
@@ -45,10 +51,17 @@
     }
 
     public Mono<Void> handleNotification(ConsentNotificationRequest consentNotificationRequest) {
-        // TODO: Need to figure out how we are going to figure out consent manager id.
-        // most probably need to have a mapping of @ncg = consent manager id
-        return validateRequest(consentNotificationRequest.getConsentRequestId())
-                .flatMap(consentRequest -> upsertConsentArtefacts(consentNotificationRequest).then());
+        if (consentNotificationRequest.getStatus() == ConsentStatus.GRANTED) {
+            // TODO: Need to figure out how we are going to figure out consent manager id.
+            // most probably need to have a mapping of @ncg = consent manager id
+            return validateRequest(consentNotificationRequest.getConsentRequestId())
+                    .flatMap(consentRequest -> upsertConsentArtefacts(consentNotificationRequest).then());
+        } else if (consentNotificationRequest.getStatus().equals(ConsentStatus.REVOKED)) {
+            return validateConsents(consentNotificationRequest.getConsentArtefacts())
+                    .flatMap(consentArtefacts -> upsertConsentArtefacts(consentNotificationRequest).then());
+        }
+        //TODO: Need to validate for all scenarios
+        return Mono.empty();
```

Comment:
Could return an error since we don't support other types now.

Reply:
But, what should be the error?